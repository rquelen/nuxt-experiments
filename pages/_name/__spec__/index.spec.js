import {shallow} from '@vue/test-utils';
import Multiselect from 'vue-multiselect';

import Name from '../index.vue';

describe('/_name', () => {
  let $route = {
    params: {},
  };

  it('should match snapshot', () => {
    // when
    const wrapper = shallow(Name, { mocks: { $route } });

    // then
    expect(wrapper.element).toBeDefined();
    expect(wrapper.element).toMatchSnapshot();
  });

  describe('title', () => {
    it('should contain default message', () => {
      // when
      const wrapper = shallow(Name, { mocks: { $route } });

      // then
      const title = wrapper.find('.title');
      expect(title.text()).toContain('Welcome');
    });

    it('should update when selectedOption data is changed', () => {
      // given
      const wrapper = shallow(Name, { mocks: { $route } });

      // when
      wrapper.setData({selectedOption: 'Bonjour'});

      // then
      const title = wrapper.find('.title');
      expect(title.text()).toContain('Bonjour');
    });

    it('should contain name in url params', () => {
      // given
      $route = {
        params: {
          name: 'nuxt',
        },
      };

      // when
      const wrapper = shallow(Name, { mocks: { $route } });

      // then
      const title = wrapper.find('.title');
      expect(title.text()).toContain('nuxt');
    });
  });

  it('should contains Multiselect with proper props', () => {
    // when
    const wrapper = shallow(Name, { mocks: { $route } });

    // then
    const selectWrapper = wrapper.find(Multiselect);

    const selectModel = selectWrapper.vm.$vnode.data.model.value;
    expect(selectModel).toBe('Welcome');

    const selectAttributes = selectWrapper.vm.$attrs;
    expect(selectAttributes.options).toEqual(['Welcome', 'Bonjour', 'Ciao']);
    expect(selectAttributes.searchable).toBe(false);
    expect(selectAttributes['allow-empty']).toBe(false);

    const selectOptions = selectWrapper.vm.$options.propsData;
    expect(selectOptions.deselectLabel).toBe('');
    expect(selectOptions.selectLabel).toBe('');
    expect(selectOptions.selectedLabel).toBe('');
  });
});
