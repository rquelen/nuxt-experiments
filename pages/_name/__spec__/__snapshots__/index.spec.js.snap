// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/_name should match snapshot 1`] = `
<section
  class="container"
>
  <h1
    class="title"
  >
    
    Welcome 
  
  </h1>
   
  <!---->
   
  <div>
    <nuxt-link
      to="/counter"
    >
      Counter
    </nuxt-link>
     
    <nuxt-link
      exact=""
      to="/"
    >
      Back to home
    </nuxt-link>
  </div>
</section>
`;
